product.py

```diff
@@ -537,9 +537,17 @@ def findXenSourceBackups():
             if os.path.exists(os.path.join(b.mount_point, '.xen-backup-partition')):
                 backup = XenServerBackup(p, b.mount_point)
                 logger.log("Found a backup: %s" % (repr(backup),))
-                if backup.version >= XENSERVER_MIN_VERSION and \
-                        backup.version <= THIS_PLATFORM_VERSION:
+                while True:
+                    if backup.version < XENSERVER_MIN_VERSION:
+                        logger.log("findXenSourceBackups: ignoring, platform too old: %s < %s" %
+                                   (backup.version, XENSERVER_MIN_VERSION))
+                        break
+                    if backup.version > THIS_PLATFORM_VERSION:
+                        logger.log("findXenSourceBackups: ignoring later platform: %s > %s" %
+                                   (backup.version, THIS_PLATFORM_VERSION))
+                        break
                     backups.append(backup)
+                    break
```

Comment:
I might be missing something, but at first sight this looks a lot like a complicated way to do what `if... elif... else`  would do. I see no way to actually loop a second time in this `while` loop.

Reply:
The loop is just here as a `try: finally:` without any exception overhead. But right, it's overkill here.